### PR TITLE
Make ./scripts a package, make scripts typecheck, remove old stuff, workflow updates

### PR DESCRIPTION
**Pull request by jakebailey**
This is a bit of a chunky PR but so far it:

- Makes the ghostbuster not full clone.
- Makes codeowners not commit, but instead use an action to do that instead. Way better for local testing.
- Drops source-map-support from the root; node does this now.
- Moves all of the script dependencies into `./scripts`, with the intent that we depend on as few packages at the repo root as possible, both for local installs but also to prevent accidental resolution.
- Removes old scripts
- Overhauls scripts that previously needed the dt-tools stuff to instead use `package.json` and other helpers.
- Makes the existing scripts typecheck, and verifies in CI

Things I want to do but haven't done:

- Get rid of the fancy formatting in codeowners. If the longest package ever changes, it'll completely reformat the file. Nobody is reading this file.
- Stop using `URL` in scripts. We should have just converted `import.meta.url` to `__filename` and used paths instead; the URL API is super annoying.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,43 +19,31 @@
...
         "remark-cli": "^11.0.0",
         "remark-gfm": "^3.0.0",
         "remark-validate-links": "^12.0.0",
-        "shelljs": "^0.8.5",
-        "source-map-support": "^0.5.21",
         "strip-ansi": "^6.0.1",
-        "typescript": "next",
-        "w3c-xmlserializer": "^2.0.0",
-        "yargs": "^17.1.1"
+        "tslint": "latest",
```
Comment (jakebailey) [inline package.json:45]: pnpm is strict, so to access this in the scripts for transition scripts, I had to add it here to make it line up.

But, we have duplicate scripts that deal with lints so not sure if they even work.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -77,3 +77,22 @@ jobs:
           TOKEN='ghp_i5wtj1l2AbpFv3OU96w6R'
           TOKEN+='On3bHOkcV2AmVY6'
           DANGER_GITHUB_API_TOKEN=$TOKEN pnpm danger ci || $( exit 0 )
+
+  scripts:
```
Comment (sandersn) [inline .github/workflows/CI.yml:81]: does this happen for every CI run?

Comment (jakebailey) [reply .github/workflows/CI.yml:81]: Yes. I can make it only happen on pushes to master or PRs that change the files, I guess, but funnily the scripts depend on other type packages so they may break our scripts (which is a good test case).

**Review by sandersn — APPROVED** [review]

File: .github/workflows/CI.yml (hunk 2 above)
Comment (sandersn) [reply .github/workflows/CI.yml:81]: Sounds fine then.

**Comment by jakebailey** [comment]
Ugh, I guess there's still more broken in dt-tools. Yay empty errors

**Review by andrewbranch — APPROVED** [review]

**Pull request merged**